File: content/09/code/js/00-start-point-including-a-person-parameter-type/test/person.test.js
const assert = require('assert')
const sinon = require('sinon')
const {Person} = require('../src/shouty')

describe('Person', () => {
  let network, networkStub
  beforeEach(() => {
    network = {
      subscribe() {},
      broadcast() {}
    }
    networkStub = sinon.spy(network)
  })

  it('subscribes to the network', () => {
    const lucy = new Person('Lucy', network, 100)
    assert(networkStub.subscribe.calledOnce)
    assert.strictEqual(networkStub.subscribe.getCall(0).args[0], lucy)
  })

  it('has a location', () => {
    const location = 100
    const lucy = new Person('Lucy', network, location)
    assert.strictEqual(lucy.location, location)
  })

  it('broadcasts shouts to the network', () => {
    const message = "Free bagels!"
    const sean = new Person('Sean', network)
    sean.shout(message)
    assert(networkStub.broadcast.calledOnce)
    assert.strictEqual(networkStub.broadcast.getCall(0).args[0], message)
  })

  it('remembers messages heard', () => {
    const message = "Free bagels!"
    const lucy = new Person('Lucy', network)
    lucy.hear(message)
    assert.deepStrictEqual(lucy.messagesHeard(), [message])
  })

  it('can be moved to a different location', () => {
    const lucy = new Person('Lucy', network, 0).at(100)
    assert.equal(lucy.location, 100)
  })
})

File: content/bdd-with-cucumber/09/code/java/01-only-run-tests-we-expect-to-pass/shouty/src/test/java/shouty/PersonTest.java
package shouty;

import org.junit.Test;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class PersonTest {
    private final Network network = mock(Network.class);

    @Test
    public void it_subscribes_to_the_network() {
        Person person = new Person("A Person:", network, 100);
        verify(network).subscribe(person);
    }

    @Test
    public void it_has_a_location() {
        Person person = new Person("A Person", network, 100);
        assertEquals(100, person.getLocation());
    }

    @Test
    public void broadcasts_shouts_to_the_network() {
        String message = "Free bagels!";
        Person sean = new Person("Sean", network, 0);
        sean.shout(message);
        verify(network).broadcast(message, sean);
    }

    @Test
    public void remembers_messages_heard() {
        String message = "Free bagels!";
        Person lucy = new Person("Lucy", network, 100);
        lucy.hear(message);
        assertEquals(asList(message), lucy.getMessagesHeard());
    }

    @Test
    public void can_be_moved_to_new_location() {
        Person lucy = new Person("Lucy", network, 0);
        assertEquals(lucy.moveTo(100).getLocation(), 100);
    }
}
